BootloaderCorePkg/Tools/GenContainer.py

```diff
@@ -626,15 +644,23 @@ def gen_container_bin (container_list, out_dir, inp_dir, key_dir = '.', tool_dir
         print ("Container '%s' was created successfully at:  \n  %s" % (container.header.signature.decode(), out_file))
 
 def adjust_auth_type (auth_type_str, key_path):
+    print('auth_type_str %s key_path %s' % (auth_type_str, key_path))
     if os.path.exists(key_path):
         sign_key_type = get_key_type(key_path)
-        auth_type, hash_type = get_auth_hash_type (sign_key_type)
+        if auth_type_str != '':
+            sign_scheme = CONTAINER._auth_to_signscheme_str[auth_type_str]
+        else:
+            # Set to default signing scheme if auth type is generated.
+            sign_scheme = 'RSA_PSS'
+        auth_type, hash_type = get_auth_hash_type (sign_key_type, sign_scheme)
         if auth_type_str and (auth_type != auth_type_str):
             print ("Override auth type to '%s' in order to match the private key type !" % auth_type)
         auth_type_str = auth_type
+
     return auth_type_str
 
 def gen_layout (comp_list, img_type, auth_type_str, out_file, key_dir, key_file):
+    print ("auth_type_str1 %s" % auth_type_str)

```

Comment:
is this for debug only ?

Reply:
Yes its for debug purposes. I would clean it up.